packages/xo-server-audit/src/index.js

```diff
@@ -293,6 +314,41 @@ class AuditXoPlugin {
       }))
   }
 
+  // See www-xo#344
+  async _uploadLastHash() {
+    const xo = this._xo
+    const chain = await xo.audit.getLastChain()
+
+    let lastHash, integrityCheckSuccess
+    if (chain !== null) {
+      const hashes = chain.hashes
+      lastHash = hashes[hashes.length - 1]
+
+      // check the integrity of all stored hashes
+      integrityCheckSuccess = await Promise.all(
+        hashes.map((oldest, key) =>
+          oldest !== lastHash
+            ? this._checkIntegrity({ oldest, newest: hashes[key + 1] })
+            : true
+        )
```

Comment:
Can you remind me why this is necessary? Why don't we just check the last hash's integrity?